nengo_gui/static/components/netgraph_conn.js

```diff
@@ -167,25 +169,40 @@ Nengo.NetGraphConnection.prototype.set_pre = function(pre) {
 /** set the item connecting to */
 Nengo.NetGraphConnection.prototype.set_post = function(post) {
     if (this.post !== null) {
-        /** if we're currently connected, disconnect */
-        var index = this.post.conn_in.indexOf(this);
-        if (index === -1) {
-            console.log('error removing in set_pre');
+        let elems = [this.post]
+        if (this.post instanceof Nengo.NetGraphConnection) {
+            elems = [this.post.pre, this.post.post]
+        }
+        for (let elem of elems) {
+            if (elem !== null) {
+                var index = elem.conn_in.indexOf(this);
+                if (index === -1) {
+                    console.log('error removing in set_pre');
+                }
+                elem.conn_in.splice(index, 1);
+            }
         }
-        this.post.conn_in.splice(index, 1);
     }
     this.post = post;
     if (this.post !== null) {
-        /** add myself to post's input connections list */
-        this.post.conn_in.push(this);
+        let elems = [this.post]
+        if (this.post instanceof Nengo.NetGraphConnection) {
+            elems = [this.post.pre, this.post.post]
+        }
```

Comment:
The preceding four lines appear twice in this function, maybe extract them into a separate function (e.g., `get_elems`)?

Reply:
I've extracted a helper function and added some documentation explaining what this code is actually doing. I've squashed my changes into the last commit.